modules/axtask/src/run_queue.rs

```diff
@@ -286,7 +286,11 @@ impl<'a, G: BaseGuard> AxRunQueueRef<'a, G> {
     ///     2. The caller must ensure that the current task is in the running state.
     ///     3. The caller must ensure that the current task is not the idle task.
     ///     4. The lock of the wait queue will be released explicitly after current task is pushed into it.
-    pub fn blocked_resched(&mut self, mut wq_guard: WaitQueueGuard) {
+    pub fn blocked_resched(
+        &mut self,
+        mut wq_guard: WaitQueueGuard,
+        curr_waiter: Arc<WaitTaskNode>,
```

Comment:
It still seems awkward, `curr_waiter` is just a wrapper for `crate::current()` in this code block.

Reply:
Here need param `waiter`, because wq_guard  have to  push back the `curr wrapper` , we can't crteate the wrapper in `blocked_resched`, because `waiter` fast delete in caller,  need use it; we can discuss  whether there is a better way

Reply:
https://github.com/arceos-org/arceos/pull/184/commits/6219d10f9339aad5770d7a371294b72155d3a560   in this commit, give a plan  remove wq_guard and waiter